Client/Frontend/Home/ActivityStreamPanel.swift

```diff
@@ -324,13 +321,42 @@ extension ActivityStreamPanel {
             let touchPoint = longPressGestureRecognizer.locationInView(self.view)
             if let indexPath = tableView.indexPathForRowAtPoint(touchPoint) {
                 if Section(indexPath.section) == .Highlights {
-                    presentContextMenu(history[indexPath.row])
+                    presentContextMenuForHighlights(history[indexPath.row])
+                } else {
+                    let topSiteCell = self.tableView.cellForRowAtIndexPath(indexPath) as! ASHorizontalScrollCell
+                    let touchPointWithinTopSiteCell = longPressGestureRecognizer.locationInView(topSiteCell.collectionView)
+                    if let indexPath = topSiteCell.collectionView.indexPathForItemAtPoint(touchPointWithinTopSiteCell) {
```

Comment:
I'm combining the context menu logic for top sites and highlights because we now need the indexpath parameter for both.